core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/services/impl/EnumOptionSourceResolver.java

```diff
@@ -0,0 +1,194 @@
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.adobe.granite.ui.components.ds.ValueMapResource;
+
+/**
+ * Implements {@link OptionSourceResolver} to facilitate extracting option datasources from enum classes.
+ */
+public class EnumOptionSourceResolver implements OptionSourceResolver {
+    private static final Logger LOG = LoggerFactory.getLogger(EnumOptionSourceResolver.class);
+    private static final String VALUE_ATTRIBUTE_KEY = "value";
+    private Class<? extends Enum<?>> clazz;
+
+    /**
+     * Default constructor
+     */
+    public EnumOptionSourceResolver() {
+    }
+
+    /**
+     * Constructor for testing
+     * @param clazz to be used instead of looking for it in {@link Bundle} array from {@link BundleContext}
+     */
+    public EnumOptionSourceResolver(Class<? extends Enum<?>> clazz) {
+        this.clazz = clazz;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Resource pathResolve(SlingHttpServletRequest request, PathParameters pathParameters) {
+        return resolve(request, pathParameters.getPath(), pathParameters);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Resource fallbackResolve(SlingHttpServletRequest request, PathParameters pathParameters) {
+        return resolve(request, pathParameters.getFallbackPath(), pathParameters);
+    }
+
+    /**
+     * Method, called by either {@link EnumOptionSourceResolver#resolve(SlingHttpServletRequest, PathParameters)} or
+     * {@link EnumOptionSourceResolver#fallbackResolve(SlingHttpServletRequest, PathParameters)} public methods.
+     * Method retrieves {@link Class} from {@link BundleContext} that should be {@link Enum} and then builds
+     * a {@link Resource} based on this Enum and {@link PathParameters}
+     * @param request {@link SlingHttpServletRequest} object
+     * @param path           should be a valid {@link String} class name.
+     * @param pathParameters represents key containing object to retrieve values from enum {@link Class}
+     * @return {@link Resource}
+     */
+    private Resource resolve(SlingHttpServletRequest request, String path, PathParameters pathParameters) {
+        if (clazz == null) {
+            setClass(path);
+        }
+
+        if (clazz == null) {
+            LOG.error("EnumOptionSourceResolver::resolve - value isn't a valid class name: {}", path);
+            return null;
+        }
+
+        if (!clazz.isEnum()) {
+            LOG.error("EnumOptionSourceResolver::resolve - class {} is not an Enum", clazz.getName());
+            return null;
+        }
+
+        if (clazz.getEnumConstants().length == 0) {
+            LOG.error("EnumOptionSourceResolver::resolve - Enum class {} doesn't have constants", clazz.getName());
```

Comment:
An edge situation but I won't see this is an error. We could have an ACS list without any items, or a tag folder currently without tags. This is rather a "content issue"